csharp_dotnetcore/17.Proactive-Messages/ProactiveBot.cs

```diff
@@ -0,0 +1,208 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Configuration;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.BotBuilderSamples
+{
+    /// <summary>
+    /// For each interaction from the user, an instance of this class is called.
+    /// This is a Transient lifetime service.  Transient lifetime services are created
+    /// each time they're requested. For each Activity received, a new instance of this
+    /// class is created. Objects that are expensive to construct, or have a lifetime
+    /// beyond the single Turn, should be carefully managed.
+    /// </summary>
+    public class ProactiveBot : IBot
+    {
+        /// <summary>The name of events that signal that a job has completed.</summary>
+        public const string JobCompleteEventName = "jobComplete";
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ProactiveBot"/> class.</summary>
+        /// <param name="accessors">The state accessors for use with the bot.</param>
+        /// <param name="endpointService">The <see cref="EndpointService"/> portion of the <see cref="BotConfiguration"/>.</param>
+        public ProactiveBot(ProactiveAccessors accessors, EndpointService endpointService)
+        {
+            StateAccessors = accessors ?? throw new ArgumentNullException(nameof(accessors));
+
+            // AppId was validated during startup.
+            AppId = endpointService.AppId;
+        }
```

Comment:
As far as I know, this is the only sample that has this. Can you add in a more detailed explication of why it's required?

Reply:
Added to property definition.

Reply:
@cleemullins , The call to the adapter's continue conversation method requires the bot's app ID.